src/transaction/log_manager.c

```diff
@@ -1729,6 +1747,19 @@ log_final (THREAD_ENTRY * thread_p)
 
   save_tran_index = LOG_FIND_THREAD_TRAN_INDEX (thread_p);
 
+  // mark and persist that transaction server with remote storage has been correctly closed
+  if (is_tran_server_with_remote_storage ())
+    {
+      log_Gl.m_metainfo.set_is_tsrs_shutdown (true);
+
+      const int res_metalog_to_file = log_write_metalog_to_file ();
+      if (res_metalog_to_file != NO_ERROR)
+	{
+	  ASSERT_ERROR ();
+	  _er_log_debug (ARG_FILE_LINE, "log_final: failed writing metalog to file\n");
+	}
```

Comment:
`log_write_metalog_to_file` is used also, as of now at least, by the TSRS trantable daemon.
There very well might appear transient errors and not all of them are fatal (see second call to write metalog in function `logpb_checkpoint_trantable`).
Even in this situation, if the flag is not properly written due to a transient error, the worst that can happen upon restart is a speedy recovery being executed.

Reply:
What kind of transient errors could prevent opening a file? If there are such errors, maybe they have to be treated.

If the metalog cannot be written at any moment, for instance in the first call in `logpb_checkpoint_trantable`, it could have more serious consequences. I think an assert or a fatal error is appropriate, even if in some circumstances, like writing the is_clean_shutdown flag, it could be accepted. It also simplifies things a bit, because the returned error no longer has to be handled (how can it be handled anyway in other way than ignoring it?).